src/custom/components/AffiliateStatusCheck/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+import { useActiveWeb3React } from 'hooks/web3'
+import NotificationBanner from 'components/NotificationBanner'
+import { useReferralAddress, useResetReferralAddress, useUploadReferralDocAndSetDataHash } from 'state/affiliate/hooks'
+import { useAppDispatch } from 'state/hooks'
+import { hasTrades } from 'utils/trade'
+import { SupportedChainId } from 'constants/chains'
+
+type AffiliateStatus = 'NOT_CONNECTED' | 'OWN_LINK' | 'ALREADY_TRADED' | 'ACTIVE' | 'UNSUPPORTED_NETWORK'
+
+const STATUS_TO_MESSAGE_MAPPING: Record<AffiliateStatus, string> = {
+  NOT_CONNECTED: 'Please connect your wallet to participate',
+  OWN_LINK: 'You followed your own referral link',
+  ALREADY_TRADED: 'You have already traded with the current account',
+  ACTIVE: 'Your affiliate link will be effective on next trade',
+  UNSUPPORTED_NETWORK: 'Only Mainnet is supported. Please change the network to participate',
+}
+
+export default function AffiliateStatusCheck() {
+  const appDispatch = useAppDispatch()
+  const resetReferralAddress = useResetReferralAddress()
+  const uploadReferralDocAndSetDataHash = useUploadReferralDocAndSetDataHash()
+  const history = useHistory()
+  const { account, chainId } = useActiveWeb3React()
+  const referralAddress = useReferralAddress()
+  const [affiliateState, setAffiliateState] = useState<AffiliateStatus | null>()
+  const [error, setError] = useState('')
+
+  // TODO: retry on error?
```

Comment:
Not for this PR, but you can add to an issue as a checkbox, or create an issue for this.

There's an util u can use: `utils/retry`

Reply:
Great, I can wrap the `hasTrades` and the `uploadReferralDocAndSetDataHash` calls individually with that retry helper. It's a small change that if it makes sense I can add it on this PR.

Should we use a common a common retry options setup such as: 

```js
{
  n: 3, 
  minWait: 1000, 
  maxWait: 3000
}
```
?

Reply:
That works!